mongorestore/filepath_test.go

```diff
@@ -112,6 +119,47 @@ func TestCreateAllIntents(t *testing.T) {
 	})
 }
 
+func TestCreateAllIntentsLongCollectionName(t *testing.T) {
+	// This tests creates intents based on the test file tree:
+	//   testdata/longcollectionname
+	//   testdata/longcollectionname/db1
+	//   testdata/longcollectionname/db1/aVery...VeryLongCollectionNameConsistingOfE%24xFO0VquRn7cg3QooSZD5sglTddU.bson
+	//   testdata/longcollectionname/db1/aVery...VeryLongCollectionNameConsistingOfE%24xFO0VquRn7cg3QooSZD5sglTddU.metadata.json
+
+	var mr *MongoRestore
+	var buff bytes.Buffer
+
+	testtype.SkipUnlessTestType(t, testtype.UnitTestType)
+
+	Convey("With a test MongoRestore", t, func() {
+		mr = newMongoRestore()
+		log.SetWriter(&buff)
+
+		Convey("running CreateAllIntents should succeed", func() {
+			ddl, err := newActualPath("testdata/longcollectionname/")
+			So(err, ShouldBeNil)
+			So(mr.CreateAllIntents(ddl), ShouldBeNil)
+			mr.manager.Finalize(intents.Legacy)
+
+			Convey("and reading the intents should show a long collection name", func() {
+				i0 := mr.manager.Pop()
+				So(i0.DB, ShouldEqual, "db1")
+				So(i0.C, ShouldEqual, longCollectionName)
+
+				Convey("with all the proper metadata + bson merges", func() {
+					So(i0.Location, ShouldNotEqual, "")
+					So(i0.MetadataLocation, ShouldNotEqual, "")
+
+					Convey("and skipped files all present in the logs", func() {
```

Comment:
Mmm yes that's a good point. I was following the structure of the test above this one, but I can remove this Convey call since it doesn't directly relate to long collection names.